mavros_extras/src/plugins/debug.cpp

```diff
@@ -0,0 +1,254 @@
+		debug_msg->value_float = debug.value;
+
+		debug_pub.publish(debug_msg);
+	}
+
+	/**
+	 * @brief Handle DEBUG_VECTOR message.
+	 * Message specification: http://mavlink.org/messages/common/#DEBUG_VECTOR
+	 * @param msg	Received Mavlink msg
+	 * @param debug	DEBUG_VECTOR msg
+	 */
+	void handle_debug_vector(const mavlink::mavlink_message_t *msg, mavlink::common::msg::DEBUG_VECT &debug)
+	{
+		auto debug_msg = boost::make_shared<mavros_msgs::Debug>();
+
+		std::string name(debug.name.data());
+		ros::Time stamp = m_uas->synchronise_stamp(debug.time_usec);
+		auto vector = Eigen::Vector3d(debug.x, debug.y, debug.z);
+
+		ROS_DEBUG_STREAM_NAMED(name, "timestamp: " << stamp
+							   << " type: DEBUG_VECTOR"
+							   << " vector: " << vector);
+
+		if (publish_debug) {
+			debug_msg->header.stamp = stamp;
+			debug_msg->name = name;
+			debug_msg->type = debug_msg->DEBUG_VECT_TYPE;
+			tf::vectorEigenToMsg(vector, debug_msg->vector);
+		}
+
+		debug_vector_pub.publish(debug_msg);
+	}
+
+	/**
+	 * @brief Handle NAMED_VALUE_FLOAT message.
+	 * Message specification: http://mavlink.org/messages/common/#NAMED_VALUE_FLOAT
+	 * @param msg	Received Mavlink msg
+	 * @param value	NAMED_VALUE_FLOAT msg
+	 */
+	void handle_named_value_float(const mavlink::mavlink_message_t *msg, mavlink::common::msg::NAMED_VALUE_FLOAT &value)
+	{
+		auto value_msg = boost::make_shared<mavros_msgs::Debug>();
+
+		std::string name(value.name.data());
+		ros::Time stamp = m_uas->synchronise_stamp(value.time_boot_ms);
+
+		ROS_DEBUG_STREAM_NAMED(name, "timestamp: " << stamp
+							   << " type: NAMED_VALUE_FLOAT"
+							   << " value: " << value.value);
+
+		if (publish_debug) {
+			value_msg->header.stamp = stamp;
+			value_msg->name = name;
+			value_msg->type = value_msg->NAMED_VALUE_FLOAT_TYPE;
+			value_msg->value_float = value.value;
+
+			named_value_float_pub.publish(value_msg);
+		}
+	}
+
+	/**
+	 * @brief Handle NAMED_VALUE_INT message.
+	 * Message specification: http://mavlink.org/messages/common/#NAMED_VALUE_INT
+	 * @param msg	Received Mavlink msg
+	 * @param value	NAMED_VALUE_INT msg
+	 */
+	void handle_named_value_int(const mavlink::mavlink_message_t *msg, mavlink::common::msg::NAMED_VALUE_INT &value)
+	{
+		auto value_msg = boost::make_shared<mavros_msgs::Debug>();
+
+		std::string name(value.name.data());
+		ros::Time stamp = m_uas->synchronise_stamp(value.time_boot_ms);
+
+		ROS_DEBUG_STREAM_NAMED(name, "timestamp: " << stamp
+							   << "type: NAMED_VALUE_INT"
+							   << "value: " << value.value);
+
+		if (publish_debug) {
+			value_msg->header.stamp = stamp;
+			value_msg->name = name;
+			value_msg->type = value_msg->NAMED_VALUE_INT_TYPE;
+			value_msg->value_float = value.value;
+
+			named_value_float_pub.publish(value_msg);
+		}
+	}
+
+	/* -*- callbacks -*- */
+
+	/**
+	 * @brief Debug callbacks
+	 * @param req	pointer to mavros_msgs/Debug.msg being published
+	 */
+	void debug_cb(const mavros_msgs::Debug::ConstPtr &req)
```

Comment:
I'm not sure, do we really need that?

Reply:
yeah. this is a bidirectional usage for the message. PX4 supports inout of these msgs atm.